sql/migrations/0001_initial.sql

```diff
@@ -132,13 +132,12 @@ CREATE TABLE poets
 
 CREATE INDEX poets_by_service_id_by_round_id ON poets (service_id, round_id);
 
-CREATE TABLE niposts
+CREATE TABLE state
```

Comment:
i think the name state is too vague, especially when the value is BLOB. what's wrong with `niposts`?

Reply:
Yes, I couldn't come up with a better name for the table.

The table is a key-value store. Before this PR it was used to store two values: `NIPost` and `NIPostBuilderState` both of which had an object serialized using the scale codec as value. Now a 3rd value is stored in this table `CommitmentATX` which stores the ID of the selected ATX for the PoST commitment.

All of these values have in common that they are determined at some point (temporarily) persisted in this table and published later as ATX. The reason for the temporary persistance is because if the node restarts it must continue to use the same values instead of determining new ones.

I just generalised the table to store any key-value pair and moved / renamed the existing functions accordingly.

Reply:
yes. i still think `nipost` is a better name than `state`. as these k/v pairs are all about nipost.

Reply:
For the table I see your point. But calling the package that contains the code `nipost` is a bit confusing imho.

`nipost.AddNIPostChallenge`, `nipost.StoreChallenge` or similar variations to me are less clear about what they do and/or have more repetition in their naming than `store.AddNiPostChallenge`.

The package handles storing of data not niposts per se.

Reply:
my thinking is that it's all storing nipost-related data. 

i am sure a new person reading the code will ask, what is in `state` table? state about what? everything?
how about calling it `kvstore`? 
and i do support having generic interface store/get/clear. it doesn't have to be the public interface. for example, you can still keep `AddNIPostChallenge()` but call a `store()` that's shared by all 3 keys.